src/bdii/htcondor-ce-provider

```diff
@@ -0,0 +1,225 @@
+"""
+share_ldif = """dn: GLUE2ShareID=%(shareid)s,%(bind_dn)s
+objectClass: GLUE2Entity
+objectClass: GLUE2Share
+objectClass: GLUE2ComputingShare
+GLUE2ShareID: %(shareid)s
+GLUE2ComputingShareServingState: production
+GLUE2ComputingShareTotalJobs: %(total_vo_jobs)s
+GLUE2ComputingShareWaitingJobs: %(idle_vo_jobs)s
+GLUE2ComputingShareRunningJobs: %(running_vo_jobs)s
+GLUE2ComputingShareComputingServiceForeignKey: %(central_manager)s
+GLUE2ComputingShareComputingEndpointForeignKey: %(endpointid)s
+GLUE2ComputingShareExecutionEnvironmentForeignKey: %(central_manager)s_standard
+GLUE2ShareServiceForeignKey: %(central_manager)s
+GLUE2ShareEndpointForeignKey: %(endpointid)s
+GLUE2ShareResourceForeignKey: %(central_manager)s_standard
+"""
+
+policy_ldif = """dn: GLUE2PolicyID=%(policyid)s,GLUE2ShareID=%(shareid)s,%(bind_dn)s
+objectClass: GLUE2Entity
+objectClass: GLUE2Policy
+objectClass: GLUE2MappingPolicy
+GLUE2PolicyID: %(policyid)s
+GLUE2PolicyScheme: org.glite.standard
+GLUE2PolicyRule: vo:%(vo)s
+GLUE2MappingPolicyShareForeignKey: %(shareid)s
+"""
+
+def main():
+    
+    # Get central_manager hostname
+    central_manager = htcondor.param.get('CENTRAL_MANAGER')
+    if not central_manager:
+        sys.stderr.write("Error: CENTRAL_MANAGER not set\n")
+        sys.exit(1)
+
+    # Get VO Names
+    vonames = htcondor.param.get('HTCONDORCE_VONames')
+    if not vonames:
+        sys.stderr.write("Error: HTCONDORCE_VONames not set\n")
+        sys.exit(1)
+    vonames = vonames.split(',')
+
+    # This is the bind DN for all entries
+    bind_dn = "GLUE2ServiceID=%s,GLUE2GroupID=resource,o=glue" % (central_manager)
+
+    # Query collector for number of CPUs and Collector ad
+    coll = htcondor.Collector(central_manager)
+    total_cores = 0
+    for ad in coll.query(htcondor.AdTypes.Startd, 'State=!="Owner"', ['Cpus']):
+        if ad.get('Cpus'):
+            total_cores += ad['Cpus']
+    coll_ad = coll.locate(htcondor.DaemonTypes.Collector)
+    version = coll_ad['CondorVersion'].split()[1]
+
+    # Print the entry for the GLUE2 Manager
+    print manager_ldif %{
+        'central_manager': central_manager, 
+        'bind_dn': bind_dn, 
+        'version': version,
+        'total_cores': total_cores,
+        }
+
+    instances = coll_ad['HostsUnclaimed']
+    # Print the entry for the GLUE2 Resource
+    print resource_ldif %{
+        'central_manager': central_manager, 
+        'bind_dn': bind_dn,
+        'instances': instances,
+        }
+
+    schedd_ads = coll.locateAll(htcondor.DaemonTypes.Schedd)
+    for schedd_ad in schedd_ads:
+
+        if not schedd_ad['AuthenticatedIdentity'] == 'computing-element@cern.ch':
+            continue
+
+        try:
+            if htcondor.SecMan().ping(schedd_ad, "READ")['AuthorizationSucceeded']:
+                state = 'ok'
+            else:
+                state = 'warning'
+            state_info = 'Authorization ping successful'
+        except Exception, e:
```

Comment:
Can we narrow the scope of this exception? I only get `RuntimeError` when I'm not authorized for a specific command or I give it a bad ad. Also, nothing seems to be done with `e` so it can be dropped